src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs

```diff
@@ -133,6 +114,105 @@ public unsafe dynamic LoadDocument(LazyStringValue keyLazy, string keyString, st
+                    return DynamicNullObject.Null;
+
+                Slice.From(_queryContext.Documents.Allocator, id, out var idSlice);
+                var references = GetCompareExchangeReferencesForItem(idSlice);
+
+                references.Add(keySlice);
+
+                var value = _documentsStorage.DocumentDatabase.ServerStore.Cluster.GetCompareExchangeValue(_queryContext.Server, keySlice);
+
+                if (value.Value == null || value.Value.TryGetMember("Object", out object result) == false)
+                    return DynamicNullObject.Null;
+
+                if (result == null)
+                    return DynamicNullObject.ExplicitNull;
+
+                if (result is BlittableJsonReaderObject bjro)
+                    return new DynamicBlittableJson(bjro);
+
+                return result;
+            }
+        }
+
+        private LazyStringValue GetSourceId(AbstractDynamicObject source)
+        {
+            if (source == null)
+                throw new ArgumentException("Cannot execute Load. Source is not set.");
+
+            var id = source.GetId() as LazyStringValue;
+            if (id == null)
+                throw new ArgumentException("Cannot execute Load. Source does not have a key.");
+
+            return id;
+        }
+
+        private bool TryGetKeySlice(LazyStringValue keyLazy, string keyString, out Slice keySlice)
+        {
+            keySlice = default;
+            if (keyLazy != null)
+            {
+                if (keyLazy.Length == 0)
+                    return false;
+
+                // we intentionally don't dispose of the scope here, this is being tracked by the references
+                // and will be disposed there.
+                Slice.External(_queryContext.Documents.Allocator, keyLazy, out keySlice);
+            }
+            else
+            {
+                if (keyString.Length == 0)
+                    return false;
+
+                // we intentionally don't dispose of the scope here, this is being tracked by the references
+                // and will be disposed there.
+                Slice.From(_queryContext.Documents.Allocator, keyString, out keySlice);
+            }
+
+            // making sure that we normalize the case of the key so we'll be able to find
+            // it in case insensitive manner
+            _queryContext.Documents.Allocator.ToLowerCase(ref keySlice.Content);
+
+            return true;
+        }
+
+        private bool TryGetCompareExchangeKeySlice(LazyStringValue keyLazy, string keyString, out Slice keySlice)
+        {
+            keySlice = default;
+            if (keyLazy != null)
+            {
+                if (keyLazy.Length == 0)
+                    return false;
+
+                var key = CompareExchangeKey.GetStorageKey(_documentsStorage.DocumentDatabase.Name, keyLazy);
+
+                Slice.From(_queryContext.Server.Allocator, key, out keySlice);
```

Comment:
Let's add same comment as above:

" // we intentionally don't dispose of the scope he ... "

Reply:
Done